core/metacling/src/TCling.cxx

```diff
@@ -1516,142 +1659,53 @@ bool TCling::LoadPCM(const std::string &pcmFileNameFullPath)
    // Avoid to call the plugin manager at all.
    R__InitStreamerInfoFactory();
 
-   if (gROOT->IsRootFile(pcmFileName)) {
+   TDirectory::TContext ctxt;
```

Comment:
@pcanal, why we need to have a RAII object restoring the state of gDirectory here, isn't the dtor of TFile supposed to restore the previous value? (I saw it is not the case because if we remove this tests start to fail...)

Reply:
The TFile does not retain the information .. and the timing would be odd (the current directory at the end of sequence of TFile creation and deletion would depend on how those are intertwined rather than being 'guaranteed' to return to the state before the first file creation).